modules/openapi-generator/src/main/java/org/openapitools/codegen/CodegenModel.java

```diff
@@ -74,6 +74,9 @@
 
     public Map<String, Object> vendorExtensions = new HashMap<String, Object>();
 
+    public boolean isJpaEntity = false;
+    public Set<String> additionalAnnotations = new TreeSet<String>();
```

Comment:
`additionalAnnotations` can also be used in future to add customised annotations to denigrated model class.

Reply:
To support additional annotation, we would also probably need to have something like `x-additional-imports` to add those annotations to the imports on the class.

Reply:
And that in turn would require some way to add additional dependencies to the POM perhaps?

Reply:
I've added a `spring-data-jpa` dependency to `pom.mustache` when user specifies the `useSpringJpa` in `configOptions`, and it adds the import to model class when that model contains a flag `isJpaEntity`